### PR TITLE
Add more scripts to publicly-cached list

### PR DESCRIPTION
**Pull request by wjordan**
This PR adds a number of new scripts to our existing list of publicly-cached scripts. This will make puzzle pages for these scripts (i.e., `ScriptsController#show`-based controller transactions) to be served with non-zero `Cache-Control: max-age` HTTP response headers, and also causes our HTTP-cache configuration to filter out all headers and cookies from the HTTP request/response other than language-related ones, allowing these to be cached in the CDN. Since `#show` is currently around 20% of our total Dashboard traffic, this could potentially improve our overall frontend server load somewhat.

As part of this change, I've changed the regex to cache all puzzles across all stages as opposed to only puzzles in stage 1. As far as I know, the only action other than `ScriptsController#show` to use the same path prefix `/s/[script]/stage/*` is `#summary_for_lesson_plans` (which is only consumed by a tool on levelbuilder to create lesson plans, and should be equally cacheable). This caching-prefix change shouldn't affect any other page behavior.

(In the future we might want to investigate configuring all of our puzzle pages to be publicly-cacheable, though this would require more testing and is less of a priority.)

### CODE REVIEW

File: cookbooks/cdo-varnish/libraries/http_cache.rb [hunk 1]
```diff
@@ -12,10 +12,28 @@ class HttpCache
   # A map from script name to script level URL pattern.
   CACHED_SCRIPTS_MAP = {
```
Comment (ashercodeorg) [inline cookbooks/cdo-varnish/libraries/http_cache.rb]: nit: This is no longer a constant, as it is mutated below.

File: cookbooks/cdo-varnish/libraries/http_cache.rb [hunk 2]
```diff
@@ -12,10 +12,28 @@ class HttpCache
...
+    20-hour
+    course1
+    course2
+    course3
+    course4
+    playlab
+    artist
+    infinity
+    iceage
+  ).each do |script_name|
```
Comment (ashercodeorg) [inline cookbooks/cdo-varnish/libraries/http_cache.rb]: nit: Any reason this list shouldn't be alphabetized?

Comment (ashercodeorg) [reply cookbooks/cdo-varnish/libraries/http_cache.rb]: Does anyone thing need to be changed around L107?

**Comment by wjordan** [comment]
Manual test on an adhoc server to verify the change, compared against production as baseline:

```
# production
$ curl -sSL https://studio.code.org/s/course2/stage/3/puzzle/2 -X GET -I | grep 'Cache-Control\|X-Cache\|X-Varnish-Cache'
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Cache-Control: no-cache="set-cookie"
X-Varnish-Cache: MISS
X-Cache: Miss from cloudfront

# adhoc w/change
$ curl -sSL https://dashboard-adhoc-hoc-cache-scripts-2.cdn-code.org/s/course2/stage/3/puzzle/2 -X GET -I | grep 'Cache-Control\|X-Cache\|X-Varnish-Cache'
Cache-Control: public,max-age=360,s-maxage=180
X-Varnish-Cache: HIT
X-Cache: Hit from cloudfront
```

File: cookbooks/cdo-varnish/libraries/http_cache.rb (hunk 2 above)
Comment (wjordan) [reply cookbooks/cdo-varnish/libraries/http_cache.rb]: The list at L107 marks certain paths in Pegasus/code.org as cached and language-keyed (as opposed to the Pegasus default, which is cached as English-only). That list might need some updating eventually (depending on whether we add new pages/paths to Code.org site), but would be unrelated to this Code Studio-specific change.

File: cookbooks/cdo-varnish/libraries/http_cache.rb (hunk 1 above)
Comment (wjordan) [reply cookbooks/cdo-varnish/libraries/http_cache.rb]: (to be pedantic) it's still a constant but a mutable one, I'll rewrite it to be immutable though.

File: cookbooks/cdo-varnish/libraries/http_cache.rb (hunk 2 above)
Comment (wjordan) [reply cookbooks/cdo-varnish/libraries/http_cache.rb]: no reason this list _should_ be alphabetized, as far as I can tell - we haven't organized script-names in alphabetical order anywhere in our codebase or in our levelbuilder interface, no need to start here.

**Pull request merged**